src/QirRuntime/build-qir-runtime.ps1

```diff
@@ -1,68 +1,86 @@
-    Copy-Item -Path (Join-Path $qirStaticPath qir *.ll) -Destination $Env:DROPS_DIR
+& (Join-Path $PSScriptRoot .. .. build set-env.ps1)
+Write-Host "##[info]Compile Q# Projects into QIR"
+$qirStaticPath = Join-Path $PSScriptRoot test QIR-static qsharp
+dotnet build $qirStaticPath -c $Env:BUILD_CONFIGURATION -v $Env:BUILD_VERBOSITY
+if ($LastExitCode -ne 0) {
+    Write-Host "##vso[task.logissue type=error;]Failed to compile Q# project at '$qirStaticPath' into QIR."
+    return
+}
+Copy-Item -Path (Join-Path $qirStaticPath qir *.ll) -Destination (Split-Path $qirStaticPath -Parent)
+# Also copy to drops so it ends up in build artifacts, for easier post-build debugging.
+Copy-Item -Path (Join-Path $qirStaticPath qir *.ll) -Destination $Env:DROPS_DIR
 
-    Write-Host "##[info]Build QIR Runtime"
-    $oldCC = $env:CC
-    $oldCXX = $env:CXX
-    $oldRC = $env:RC
+Write-Host "##[info]Build QIR Runtime"
+$oldCC = $env:CC
+$oldCXX = $env:CXX
+$oldRC = $env:RC
 
-    $clangTidy = ""
+$clangTidy = ""
 
-    if (($IsMacOS) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Darwin"))))
-    {
-        Write-Host "On MacOS build QIR Runtim using the default C/C++ compiler (should be AppleClang)"
-    }
-    elseif (($IsLinux) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Lin"))))
-    {
-        Write-Host "On Linux build QIR Runtime using Clang"
-        $env:CC = "/usr/bin/clang-11"
-        $env:CXX = "/usr/bin/clang++-11"
-        $env:RC = "/usr/bin/clang++-11"
-        $clangTidy = "-DCMAKE_CXX_CLANG_TIDY=clang-tidy-11"
-    }
-    elseif (($IsWindows) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Win"))))
-    {
-        Write-Host "On Windows build QIR Runtime using Clang"
-        $env:CC = "C:\Program Files\LLVM\bin\clang.exe"
-        $env:CXX = "C:\Program Files\LLVM\bin\clang++.exe"
-        $env:RC = "C:\Program Files\LLVM\bin\clang++.exe"
-        $llvmExtras = (Join-Path $PSScriptRoot "externals/LLVM")
-        $env:PATH += ";$llvmExtras"
-    } else {
-        Write-Host "##vso[task.logissue type=error;]Failed to identify the OS. Will use default CXX compiler"
-    }
+if (($IsMacOS) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Darwin"))))
+{
+    Write-Host "On MacOS build QIR Runtim using the default C/C++ compiler (should be AppleClang)"
+}
+elseif (($IsLinux) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Lin"))))
+{
+    Write-Host "On Linux build QIR Runtime using Clang"
+    $env:CC = "/usr/bin/clang-11"
+    $env:CXX = "/usr/bin/clang++-11"
+    $env:RC = "/usr/bin/clang++-11"
+    $clangTidy = "-DCMAKE_CXX_CLANG_TIDY=clang-tidy-11"
+}
+elseif (($IsWindows) -or ((Test-Path Env:AGENT_OS) -and ($Env:AGENT_OS.StartsWith("Win"))))
+{
+    Write-Host "On Windows build QIR Runtime using Clang"
+    $env:CC = "C:\Program Files\LLVM\bin\clang.exe"
+    $env:CXX = "C:\Program Files\LLVM\bin\clang++.exe"
+    $env:RC = "C:\Program Files\LLVM\bin\clang++.exe"
```

Comment:
(If you have time and desire )
:-\ Absolute paths... I understand that they were here prior to this PR... 
If you have time and desire then you may want to consider providing the path as an optional command line arg or env var. (in case the user has multiple versions installed and wants to run against a specific version)

Reply:
It's also worth noting that the intent of using PowerShell in these repos is for cross-platform compatibility, such that contributors compiling on non-Windows platforms won't have a Program Files folder.

Reply:
@cgranade, I believe this particular fragment is Windows-specific, so the users will have the "Program Files", however your comment is still applicable. The users may have installed clang to the non-default location (outside of "Program Files").

Reply:
Agreed, absolute paths are generally bad. In this case though, the script updates these variables based on the detected platform, and is already used in the build pipeline to run on OS X and Ubuntu in addition to Windows. Because of the way chocolatey installs LLVM on Windows and how we use the setup script to deploy dependencies, we need to use this full path or else the exe won't be found on the pipeline.

Reply:
As an alternative, I could avoid needing to specify paths at all by updating the setup script to add the installed LLVM tools to the path. Let me try that approach; should hopefully make for a cleaner script.